celesta-maven-plugin/src/main/java/ru/curs/celesta/plugin/maven/CursorGenerator.java

```diff
@@ -314,8 +320,40 @@ private static String calcClassName(GrainElement ge, String sourceFileNamePrefix
         return Arrays.asList(grainName, objectName);
     }
 
-    private static List<FieldSpec> buildFields(DataGrainElement ge) {
-        Map<String, ? extends ColumnMeta> columns = ge.getColumns();
+    private static TypeSpec buildCursorColumnsAsInnerClass(DataGrainElement dge, ClassName columnsClassType) {
+
+        TypeSpec.Builder builder = TypeSpec.classBuilder(columnsClassType)
+                .addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+
+        dge.getColumns().entrySet().stream()
+                .filter(e -> !BinaryColumn.CELESTA_TYPE.equals(e.getValue().getCelestaType()))
+                .map(e -> {
+                    final String columnName = e.getKey();
+                    final TypeName columnType =
+                            ParameterizedTypeName.get(ColumnMeta.class, e.getValue().getJavaClass());
+                    return FieldSpec.builder(columnType, columnName,
+                                             Modifier.PUBLIC, Modifier.FINAL)
+                            .initializer("($T) meta().getColumns().get($S)", columnType, columnName);
+                 })
+                .map(FieldSpec.Builder::build)
+                .forEach(builder::addField);
+
+        MethodSpec privateConstructor = MethodSpec.constructorBuilder()
+                .addModifiers(Modifier.PRIVATE)
+                .build();
+        builder.addMethod(privateConstructor);
+
+        return builder.build();
+    }
+
+    private static FieldSpec buildColumnsField(TypeName columnsClassType) {
+        return FieldSpec.builder(columnsClassType, "COLUMNS", Modifier.PUBLIC, Modifier.FINAL)
```

Comment:
I made the field name **COLUMNS** capitalized in order to escape probable naming conflicts when there's a column named **column**.